scripts/devel/build_and_install.py

```diff
@@ -23,28 +23,25 @@
 from ros_buildfarm.argument import add_argument_build_tool_args
 from ros_buildfarm.argument import add_argument_ros_version
 from ros_buildfarm.argument import add_argument_run_abichecker
+from ros_buildfarm.common import get_package_condition_context
 from ros_buildfarm.common import get_packages_in_workspaces
 from ros_buildfarm.common import Scope
 from ros_buildfarm.workspace import call_build_tool
 from ros_buildfarm.workspace import clean_workspace
 from ros_buildfarm.workspace import ensure_workspace_exists
 
 
-def call_abi_checker(workspace_root, ros_version, env):
+def call_abi_checker(workspace_root, rosdistro_name, env):
     import rosdistro
+    index = rosdistro.get_index(rosdistro.get_index_url())
 
-    condition_context = {}
-    condition_context['ROS_DISTRO'] = env['ROS_DISTRO']
-    condition_context['ROS_VERSION'] = ros_version
-    condition_context['ROS_PYTHON_VERSION'] = \
-        (env or os.environ).get('ROS_PYTHON_VERSION')
+    condition_context = get_package_condition_context(index, rosdistro_name)
```

Comment:
There is a slight behavior change here - `ROS_PYTHON_VERSION` used to come from the environment vars, but now comes from the rosdistro index (same as bloom).

Reply:
That means a user can't override the Python version anymore when running the job / script locally. How about overwriting the value in the dict returned by the function in case the env var is set? That would avoid the behavior change.

Reply:
Should I put that functionality in the common function, or do you think it should be done on a case-by-case basis?

Reply:
Hm, thinking about this... a user might also have a ROS distro sourced which set the env var and then wants to run a build for a different distro...

But wait: this is happening in `call_abi_checker` which only collects the package names of the workspace and checks if they are released. Why do we need to evaluate conditions at all here?

Reply:
Appears that `get_packages_in_workspace` implicitly performs conditional evaluation: https://github.com/ros-infrastructure/ros_buildfarm/blob/8949886a16609c78ca8ebafc47fc89a53ff7bfad/ros_buildfarm/common.py#L572-L573

Since we only want the package names, I'd suppose we can just give it an empty dict.

Reply:
We could even modify the API if you pass `None` to not evaluate the conditions at all.

Reply:
Should be addressed by #762